### PR TITLE
Ensure target is attached when executing js for init

### LINKED ISSUE #101: Uncaught TypeError in contextMenuConnector.js
We are using the context menu in our application and get the following error:

Uncaught TypeError: Cannot read property '$contextMenuConnector' of null
    at Object.init (contextMenuConnector.js:10)
    at Object.eval (eval at tt (client-59E69D05DF6B41AFC722309793280A0E.cache.js:980), <anonymous>:3:41)
    at tt (client-59E69D05DF6B41AFC722309793280A0E.cache.js:980)
    at st (client-59E69D05DF6B41AFC722309793280A0E.cache.js:937)
    at qt (client-59E69D05DF6B41AFC722309793280A0E.cache.js:562)
    at Yq (client-59E69D05DF6B41AFC722309793280A0E.cache.js:479)
    at ur.vr [as W] (client-59E69D05DF6B41AFC722309793280A0E.cache.js:988)
    at oA (client-59E69D05DF6B41AFC722309793280A0E.cache.js:870)
    at xk (client-59E69D05DF6B41AFC722309793280A0E.cache.js:744)
    at Ik.Jk [as W] (client-59E69D05DF6B41AFC722309793280A0E.cache.js:988)
    at oA (client-59E69D05DF6B41AFC722309793280A0E.cache.js:870)
    at $q (client-59E69D05DF6B41AFC722309793280A0E.cache.js:983)
    at or.pr [as D] (client-59E69D05DF6B41AFC722309793280A0E.cache.js:988)
    at Ej (client-59E69D05DF6B41AFC722309793280A0E.cache.js:396)
    at Sq (client-59E69D05DF6B41AFC722309793280A0E.cache.js:984)
    at Tq (client-59E69D05DF6B41AFC722309793280A0E.cache.js:968)
    at et.gt [as qb] (client-59E69D05DF6B41AFC722309793280A0E.cache.js:988)
    at TA.UA [as K] (client-59E69D05DF6B41AFC722309793280A0E.cache.js:988)
    at XMLHttpRequest.<anonymous> (client-59E69D05DF6B41AFC722309793280A0E.cache.js:572)
    at sb (client-59E69D05DF6B41AFC722309793280A0E.cache.js:421)
    at vb (client-59E69D05DF6B41AFC722309793280A0E.cache.js:856)
    at XMLHttpRequest.<anonymous> (client-59E69D05DF6B41AFC722309793280A0E.cache.js:585)

The problem is that init is getting called with target = null. I can't figure out why this happens, the call stack only shows obfuscated code. Maybe a null check could be included here?

Tested with Vaadin 14.0.0 and Chrome

### PR DESCRIPTION
**Pull request by alvarezguille**
Prevent client-side exception when component is initialized for a non-existing target.

Issue could be reproduced by attaching and detaching the target on a single server roundtrip.

Fixes #101

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-context-menu-flow/103)

### CODE REVIEW

File: vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java [hunk 1]
```diff
@@ -370,8 +370,10 @@ private void resetContent() {
 
     private void onTargetAttach(UI ui) {
         ui.getInternals().addComponentDependencies(ContextMenu.class);
-        ui.getPage().executeJs(
-                "window.Vaadin.Flow.contextMenuConnector.init($0)", target);
+        ui.beforeClientResponse(target, context -> {
```
Comment (pekam) [inline vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java]: I'm not sure how big of a deal this is for such a corner case, but each time the attach-detach combo is executed, it registers a new execution, which is not actually cancelled, but postponed until the target remains attached. So in the test view, spamming the `attach-detach`-button can bloat the server memory with these executions. Once the target is actually attached, it will flush all of those.

That could be improved by always running the execution, but checking if the target is actually attached at that point. To always run instead of postpone the execution, you can use `ui` instead of `target` as the first parameter.
```java
ui.beforeClientResponse(ui, context -> {
    if (target.getUI().isPresent()) {
        ui.getPage().executeJs(
                "window.Vaadin.Flow.contextMenuConnector.init($0)",
                target);
    }
});
```

File: vaadin-context-menu-flow-integration-tests/src/test/java/com/vaadin/flow/component/contextmenu/it/ContextMenuPageIT.java [hunk 2]
```diff
@@ -218,6 +218,14 @@ public void detachTarget_attachTarget_rightClickTarget_menuOpens() {
         verifyOpened();
     }
 
+    @Test
+    public void detachedTarget_attachDetachTarget_noErrors() {
+        clickElementWithJs("detach-target");
+
+        clickElementWithJs("attach-detach-target");
+        checkLogsForErrors();
```
Comment (pekam) [inline vaadin-context-menu-flow-integration-tests/src/test/java/com/vaadin/flow/component/contextmenu/it/ContextMenuPageIT.java:226]: This resulted in false-positive result for me, when running without the fix.

**Review by pekam — CHANGES_REQUESTED** [review]

**Review by alvarezguille — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-context-menu-flow/103)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @alvarezguille and @vaadin-bot)

___
*[vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java, line 373 at r1](https://reviewable.io/reviews/vaadin/vaadin-context-menu-flow/103#-Lp89p9Tt_TNv-F_32V8-r1-373:-LpCEsni8BTZc0ajwsZE:b-p4psi2) ([raw file](https://github.com/vaadin/vaadin-context-menu-flow/blob/2edde26512450097819b7b18c2aa738276eeecae/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java#L373)):*
<details><summary><i>Previously, pekam (Pekka Maanpää) wrote…</i></summary><blockquote>

I'm not sure how big of a deal this is for such a corner case, but each time the attach-detach combo is executed, it registers a new execution, which is not actually cancelled, but postponed until the target remains attached. So in the test view, spamming the `attach-detach`-button can bloat the server memory with these executions. Once the target is actually attached, it will flush all of those.

That could be improved by always running the execution, but checking if the target is actually attached at that point. To always run instead of postpone the execution, you can use `ui` instead of `target` as the first parameter.
```java
ui.beforeClientResponse(ui, context -> {
    if (target.getUI().isPresent()) {
        ui.getPage().executeJs(
                "window.Vaadin.Flow.contextMenuConnector.init($0)",
                target);
    }
});
```
</blockquote></details>

Done. 
Single registration now

File: vaadin-context-menu-flow-integration-tests/src/test/java/com/vaadin/flow/component/contextmenu/it/ContextMenuPageIT.java (hunk 2 above)
Comment (pekam) [reply vaadin-context-menu-flow-integration-tests/src/test/java/com/vaadin/flow/component/contextmenu/it/ContextMenuPageIT.java:226]: ...only in bower mode. Since it works with npm, it's fine.

File: vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java [hunk 3]
```diff
@@ -370,8 +371,15 @@ private void resetContent() {
 
     private void onTargetAttach(UI ui) {
         ui.getInternals().addComponentDependencies(ContextMenu.class);
-        ui.getPage().executeJs(
-                "window.Vaadin.Flow.contextMenuConnector.init($0)", target);
+        if (targetBeforeClientResponseRegistration == null) {
+            targetBeforeClientResponseRegistration = ui
```
Comment (pekam) [inline vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java:379]: This should be removed and set to null in `setTarget`, to cancel the `connector.init` call for the previous target in case the target is changed multiple times during the same roundtrip.

**Review by alvarezguille — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-context-menu-flow/103)* status: 0 of 6 files reviewed, 3 unresolved discussions (waiting on @alvarezguille, @pekam, and @vaadin-bot)

___
*[vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java, line 375 at r2](https://reviewable.io/reviews/vaadin/vaadin-context-menu-flow/103#-Lp89p9Tt_TNv-F_32V8-r2-375:-LpU--Jh0aVwLif82AFD:bb6un1j) ([raw file](https://github.com/vaadin/vaadin-context-menu-flow/blob/5ab606f5baa5708922b38c67e405155039ed2945/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java#L375)):*
<details><summary><i>Previously, pekam (Pekka Maanpää) wrote…</i></summary><blockquote>

This should be removed and set to null in `setTarget`, to cancel the `connector.init` call for the previous target in case the target is changed multiple times during the same roundtrip.
</blockquote></details>

Done

**Review by pekam — APPROVED** [review]
LGTM 👌
